src/javascript-extract.js

```diff
@@ -33,20 +34,42 @@ function extractConcatenatedStrings(value, allTokens, index) {
 }
 
 
-function getGettextEntriesFromScript(script) {
+function getGettextEntriesFromScript(script, parser = 'auto') {
+
   const allTokens = [];
 
-  const ACORN_OPTIONS = {
-    ecmaVersion: 10,
-    sourceType: 'module',
-    locations: true,
-    onToken: allTokens,
-    plugins: {
-      stage3: true,
-    },
-  };
+  switch (parser) {
+  case 'auto':
+    try {
+      return getGettextEntriesFromScript(script, 'acorn');
+    } catch (e) {
```

Comment:
you should log something so that the user sees that we just switched the parser.